builtin/logical/totp/path_keys.go

```diff
@@ -0,0 +1,400 @@
+package totp
+
+import (
+	"bytes"
+	"encoding/base32"
+	"encoding/base64"
+	"fmt"
+	"image/png"
+	"net/url"
+	"strconv"
+	"strings"
+
+	"github.com/hashicorp/vault/logical"
+	"github.com/hashicorp/vault/logical/framework"
+	otplib "github.com/pquerna/otp"
+	totplib "github.com/pquerna/otp/totp"
+)
+
+func pathListKeys(b *backend) *framework.Path {
+	return &framework.Path{
+		Pattern: "keys/?$",
+
+		Callbacks: map[logical.Operation]framework.OperationFunc{
+			logical.ListOperation: b.pathKeyList,
+		},
+
+		HelpSynopsis:    pathKeyHelpSyn,
+		HelpDescription: pathKeyHelpDesc,
+	}
+}
+
+func pathKeys(b *backend) *framework.Path {
+	return &framework.Path{
+		Pattern: "keys/" + framework.GenericNameRegex("name"),
+		Fields: map[string]*framework.FieldSchema{
```

Comment:
Please add a boolean field (`exported` probably) that defaults to true and that controls whether or not a key that is generated is actually returned as a URL/QR code. The use-case is if you want to give one Vault user access to fetch a code, another access to validate the codes, but you don't ever actually want the key to be exposed.

A neat enhancement in this case might be to have the algorithm default to SHA256 if this is false, because if you know the key is never used anywhere else, you don't have to worry about compatibility issues.